src/Adapter/Cart/QueryHandler/GetCartInformationHandler.php

```diff
@@ -120,16 +122,23 @@ public function handle(GetCartInformation $query): CartInformation
 
         $legacySummary = $cart->getSummaryDetails(null, true);
         $addresses = $this->getAddresses($cart);
+        $cartRules = $this->extractCartRulesFromLegacySummary($legacySummary, $currency);
+        $hasFreeShippingCartRule = false;
+        foreach ($legacySummary['discounts'] as $discount) {
+            if ((bool) $discount['free_shipping'] && $discount['name'] != self::FREE_SHIPPING_NAME) {
```

Comment:
I don't think you can rely on the value of a cart rule name, this can be edited in the back office So it's not a reliable information

Reply:
Ah, this one is tricky,  when FREE SHIPPING for order is toggled in the UI, it actually creates it always with the same name _Free Shipping_  (so it is hard coded in code): https://github.com/PrestaShop/PrestaShop/blob/d3099d26dff606fa4d38bcecdfccddfdeb1f80f5/src/Adapter/Cart/CommandHandler/SetFreeShippingToCartHandler.php#L114-L137 
The only other unique info is the code BO_ORDER (well it can also be changed in the the back office creating new cart rule). So have no idea, how make it more reliable!?

Reply:
I see a `free_shipping` SQL column into Cart Rule SQL table. Can we check what is inside this column ? it looks like a reliable information

Reply:
Looking to select * from ps_cart_rule results I don't see possibility to make it reliable without adding one more not changeable db column which defines what this cart rule came from toggling the Order FREE shipping button. But maybe we should think out of this box and search for different solution?

Reply:
@matks @jolelievre I think I'll try leave only ` if ((bool) $discount['free_shipping']` here because it makes sense. And will search a way to modify javascripts for page to render as expected

Reply:
@jolelievre @matks Thank you both, provided a cleaner solution - no longer relying on the Discount/Cart Rule name!